src/app/[locale]/incident/contact/components/IncidentContactForm.tsx

```diff
@@ -59,12 +68,42 @@ const IncidentContactForm = () => {
       sharing_allowed: values.sharing_allowed,
     })
 
-    setLastCompletedStep(3)
+    addVisitedStep(FormStep.STEP_3_CONTACT)
     addOneStep()
 
-    router.push('/incident/summary')
+    router.push(steps[FormStep.STEP_4_SUMMARY])
   }
 
+  useEffect(() => {
+    if (navToSummary) {
+      updateForm({
+        ...formState,
+        email: form.getValues('email'),
+        phone: form.getValues('phone'),
+        sharing_allowed: form.getValues('sharing_allowed'),
+      })
+
+      goToStep(FormStep.STEP_4_SUMMARY)
+      router.push(steps[FormStep.STEP_4_SUMMARY])
+      onNavToSummary(false)
+    }
+  }, [navToSummary])
+
+  useEffect(() => {
+    if (goBack) {
+      updateForm({
```

Comment:
Deze updateForm() logica zou je dan aanspreken door gewoon een call te doen vanuit het FormProgress component naar de betreffende form.submit() functie (die als goed is gewoon dan de juiste handleSubmit() functie in betreffend component aanroept).